### PR TITLE
Remove var dependency

### PR DESCRIPTION
**Pull request by at7**
Remove StrainSlice dependency from Feature and add feature_Slice call to variation API where it is required. Remove test which was testing that feature_Slice returns a StrainSlice.

### CODE REVIEW

**Comment by at7** [comment]
Yes. The change is fine for the variation API.

File: modules/t/strainSlice.t [hunk 1]
```diff
@@ -43,15 +43,6 @@ my $strain_slice = Bio::EnsEMBL::Variation::StrainSlice->new(
 
 my $vf_adaptor = $vdba->get_VariationFeatureAdaptor;
```
Comment (tgrego) [inline modules/t/strainSlice.t]: This can be removed as it is not used anymore.

Comment (at7) [reply modules/t/strainSlice.t]: It is used further down line 83.

Comment (tgrego) [reply modules/t/strainSlice.t]: indeed, my bad.

**Pull request merged**